File: app_watcher/src/main/java/com/tongchen/twatcher/mzitu/model/entity/MZiTuCategory.java
package com.tongchen.twatcher.mzitu.model.entity;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by TongChen at 18:26 on 2018/12/27.
 * <p>
 * Description: 妹子图分类
 */
public class MZiTuCategory implements Parcelable {

    //  显示的名称
    private String mCategoryName;
    //  请求用的参数
    private String mRequestName;

    public MZiTuCategory(Builder builder) {
        mCategoryName = builder.categoryName;
        mRequestName = builder.requestName;
    }

    public String getCategoryName() {
        return mCategoryName;
    }

    public String getRequestName() {
        return mRequestName;
    }

    protected MZiTuCategory(Parcel in) {
        this.mCategoryName = in.readString();
        this.mRequestName = in.readString();
    }

    public static final Parcelable.Creator<MZiTuCategory> CREATOR = new Parcelable.Creator<MZiTuCategory>() {
        @Override
        public MZiTuCategory createFromParcel(Parcel source) {
            return new MZiTuCategory(source);
        }

        @Override
        public MZiTuCategory[] newArray(int size) {
            return new MZiTuCategory[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.mCategoryName);
        dest.writeString(this.mRequestName);
    }

    public static class Builder {

        private String categoryName;
        private String requestName;

        public Builder categoryName(String categoryName) {
            this.categoryName = categoryName;
            return this;
        }

        public Builder requestName(String requestName) {
            this.requestName = requestName;
            return this;
        }

        public MZiTuCategory build() {
            return new MZiTuCategory(this);
        }
    }
}

File: app_watcher/src/main/java/com/tongchen/twatcher/mzitu/model/entity/MZiTuCategory.kt
package com.tongchen.twatcher.mzitu.model.entity

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

/**
 * @author TongChen
 * @date 2019/11/28  16:11
 * <p>
 * Desc: 妹子图分类
 */
@Parcelize
data class MZiTuCategory(
        //  显示的名称
        val categoryName: String? = "",
        //  请求用的参数
        val requestName: String? = "") : Parcelable {

    constructor(builder: Builder) : this(builder.categoryName, builder.requestName) {

    }

    class Builder {
        var categoryName: String? = ""
        var requestName: String? = ""

        fun categoryName(categoryName: String?): Builder {
            this.categoryName = categoryName
            return this
        }

        fun requestName(requestName: String?): Builder {
            this.requestName = requestName
            return this
        }

        fun build(): MZiTuCategory {
            return MZiTuCategory(this)
        }
    }
}